File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int currMaxProfit = 0;
        int lowestPrice = prices[0];
        int right = 1;
        while (right < prices.length) {
            currMaxProfit = Math.max(currMaxProfit, (prices[right] - lowestPrice));
            if (lowestPrice > prices[right]) {
                lowestPrice = prices[right];
            }
            right++;
        }
        return currMaxProfit;
    }
}

File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let currMaxProfit = 0; currLowestPrice = prices[0]
    for (let i = 1; i < prices.length; i++) {
        if (prices[i] > currLowestPrice) {
            currMaxProfit = Math.max(currMaxProfit, prices[i] - currLowestPrice)
        } else if (prices[i] < currLowestPrice) {
            currLowestPrice = prices[i]
        }
    }
    return currMaxProfit
};